talk/basicconcepts/scopesnamespaces.tex

```diff
@@ -103,24 +103,25 @@
           int a; // no clash
         }
       }
-      int f() {
+      void f() {
         n::a = 2;
       }
     \end{cppcode*}
     \columnbreak
     \begin{cppcode*}{gobble=2,firstnumber=14}
       namespace p {
-        int f() {
+        void f() {
           p::a = 2;
           a = 2;  //same as above
+          ::a = 3;
           p::inner::a = 4;
           inner::a = 4;
           n::a = 5;
         }
       }
       using namespace p::inner;
-      int g() {
-        a = 3; // using p::inner
+      void g() {
+        a = 3; // error: ambiguous
```

Comment:
I don't know if this is the best way to present this example, but this line is definitely a compile error. Maybe it's also a good idea to present it, so people know that we can get into ambiguities with if we pull several namespaces into each other.

Reply:
At least, the slide was buggy and this fixes it. I kind of like the fact that we immediately show the drawbacks of using `using` too much.